dispatch_mem_profiler.py

```diff
@@ -0,0 +1,115 @@
+import functools
+import torch
+
+import torch
+import torch.nn as nn
+from torch.utils._pytree import tree_map, tree_flatten
+from typing import List, Any
+from numbers import Number
+from collections import defaultdict
+from torch.utils._python_dispatch import push_torch_dispatch_mode, TorchDispatchMode
+aten = torch.ops.aten
+
+MB = 1024 * 1024.0
+
+operator_names = defaultdict(int)
+mem_usage = defaultdict(float)
+markers = defaultdict(int)
+series = {}
+
+def normalize_tuple(x):
+    if not isinstance(x, tuple):
+        return (x,)
+    return x
+
+
+class MemoryProfileDispatchMode(TorchDispatchMode):
+    def __init__(self, verbose=False):
+        self.verbose = verbose
+    
+    def __torch_dispatch__(self, func, types, args=..., kwargs=None):
+
+        global mem_usage, operator_names
```

Comment:
You don't need this since you only write into these dictionaries.
Same for the other functions below except the ones that do set `mem_usage`.